src/main/java/com/kenshoo/xlsx2csv/XlsxToCsvConverter.java

```diff
@@ -55,10 +57,11 @@ public XlsxToCsvConverter(String delimiter) {
      * @throws IOException if an I/O error occurs.
      */
     public void convert(InputStream inputStream, FileOutputStream fileOutputStream) throws IOException {
-        final Iterator<Row> rowIterator = getRowIterator(inputStream);
-        if (!rowIterator.hasNext()) {
+        final Optional<Iterator<Row>> optionalRowIterator = getRowIterator(inputStream);
+        if (!optionalRowIterator.isPresent() || !optionalRowIterator.get().hasNext()) {
```

Comment:
Can replace most of this with
`rowIterator.orElseThrow(NoSheetFoundException)`
Can handle the empty sheet `!hasnext()` with another exception.
I think log is not enough (but using SLF4J is good)